app/controllers/quotes_controller.rb

```diff
@@ -1,14 +1,29 @@
 class QuotesController < ApplicationController
+  before_action :authenticate_user!
   def show
     @quote = Quote.find params[:id]
   end
 
+  def edit
+    @quote = current_user.quotes.find params[:id]
+  end
+
+  def update
+    @quote = current_user.quotes.find params[:id]
+    if @quote.update quote_params
+      redirect_to @quote
+    else
+      render :edit
+    end
+  end
+
   def new
     @quote = Quote.new
   end
 
   def create
     @quote = Quote.new quote_params
+    @quote.user = current_user
```

Comment:
I will go ahead and change the association to be created_by and rename author to quoted_person as that is definitely much more clear.  Will try to be more precise in such a manner moving forward.